src/compiler/checker.ts

```diff
@@ -17834,6 +17834,23 @@ namespace ts {
             const links = getNodeLinks(node.expression);
             if (!links.resolvedType) {
                 links.resolvedType = checkExpression(node.expression);
+
+                if (isPropertyDeclaration(node.parent) && isClassLike(node.parent.parent)) {
+                    const container = getEnclosingBlockScopeContainer(node);
+                    let current = container;
+                    let containedInIterationStatement = false;
+                    while (current && !nodeStartsNewLexicalEnvironment(current)) {

```

Comment:
We're doing something similar already in `checkNestedBlockScopedBinding`. Is there overlap here that could be refactored into a separate method?

Reply:
Refactored this into `getEnclosingIterationStatement`